ca_cdk_constructs/eks/eks_cluster_integration.py

```diff
@@ -0,0 +1,82 @@
+from aws_cdk import CfnOutput, Fn, Stack
+from aws_cdk.aws_ec2 import IVpc
+from aws_cdk.aws_eks import Cluster, OpenIdConnectProvider
+from aws_cdk.aws_iam import Role, AccountRootPrincipal
+import aws_cdk.custom_resources as cr
+from constructs import Construct
+
+
+class EksClusterIntegration(Construct):
+    """Looks up an existing EKS cluster by name and provisions the kubectl IAM role needed for deployments to the cluster."""
+
+    def __init__(self, scope: Construct, id: str, vpc: IVpc, cluster_name: str):
+        super().__init__(scope, id)
+
+        self.role = Role(
+            self,
+            "KubernetesAuthRole",
+            assumed_by=AccountRootPrincipal(),
+            description="Role used by the AWS CDK project to authenticate with Kubernetes",
+        )
+
+        describe_cluster_sdk_call = cr.AwsSdkCall(
+            service="EKS",
+            action="describeCluster",
+            physical_resource_id=cr.PhysicalResourceId.of("ClusterOidcProviderLookup"),
```

Comment:
this should be called differently but it's a pain to update existing stacks that relies on it

Reply:
do you mean `ClusterOidcProviderLookup` ?